mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -310,11 +439,23 @@ object StringIndexerModel extends MLReadable[StringIndexerModel] {
     override def load(path: String): StringIndexerModel = {
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
       val dataPath = new Path(path, "data").toString
-      val data = sparkSession.read.parquet(dataPath)
-        .select("labels")
-        .head()
-      val labels = data.getAs[Seq[String]](0).toArray
-      val model = new StringIndexerModel(metadata.uid, labels)
+
+      val (majorVersion, minorVersion) = majorMinorVersion(metadata.sparkVersion)
+      val labelsArray = if (majorVersion < 2 || (majorVersion == 2 && minorVersion <= 3)) {
```

Comment:
I'm confused by this logic -- are we expecting people to use the MLlib code with different version of Spark here?

Reply:
This is for loading old StringIndexerModel saved by previous Spark. Previous model has `labels`, but new model has `labelsArray`.

Reply:
But this needs to change for Spark 3.0 now.

Reply:
Thanks. I did it.